src/libraries/System.Private.CoreLib/src/System/Threading/ThreadLocal.cs

```diff
@@ -165,7 +168,7 @@ protected virtual void Dispose(bool disposing)
         {
             int id;
 
-            lock (s_idManager)
+            lock (s_idManagerLock)
```

Comment:
Do we already have C# support for Lock or it just uses Lock's object header?

Reply:
There is C# support for the `lock` keyword on a `Lock` type, here's the [speclet](https://github.com/dotnet/csharplang/blob/main/proposals/lock-object.md)

Reply:
assume we are making sure that the IL generated is correct for this change?

Reply:
There are some basic tests for the behavior [here](https://github.com/dotnet/runtime/blob/1e0c24330c4074de89d13bca21b3b670f051f6eb/src/libraries/System.Threading/tests/LockTests.cs#L16). Will check the IL as well.

Reply:
yeah my point that it's not clear whether it kicks in or not 🙂 since it will sort of work in both cases, although, I presume it may work incorrectly when you mix both `lock ()` syntax with explicit TryEnter calls

Reply:
Yea it's a bit of a tradeoff. Currently the C# compiler would warn if Lock is casted to object, so it wouldn't be easy to mix both.